From a57cd1dfa32ddf0c2c0de83d8d5132dbffae0667 Mon Sep 17 00:00:00 2001
From: Bruce Ashfield <bruce.ashfield@gmail.com>
Date: Thu, 10 Nov 2022 15:39:31 -0500
Subject: [PATCH] build: partially revert kbuild: add phony targets to ./Kbuild

As part of commit ed7ceac157c27bdc64e79a322 [kbuild: add phony targets to ./Kbuild]
a loop has been created between prepare0 and prepare.

prepare depends on prepare0, and prepare0 now explictity calls "make
prepare", and our loop continues until an eventual hang.

We keep the rest of the commit, but drop the explicit call to "make
prepare", and we restore the previous functionality which we use to
recreate the kernel build support applications on the target.

Signed-off-by: Bruce Ashfield <bruce.ashfield@gmail.com>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index d4a0c743a9b5..41d70d77dbf1 100644
--- a/Makefile
+++ b/Makefile
@@ -1267,7 +1267,7 @@ archprepare: outputmakefile archheaders archscripts scripts include/config/kerne
 
 prepare0: archprepare
 	$(Q)$(MAKE) $(build)=scripts/mod
-	$(Q)$(MAKE) $(build)=. prepare
+	$(Q)$(MAKE) $(build)=.
 
 # All the preparing..
 prepare: prepare0
-- 
2.19.1

